framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/DbNamingConventionRewriterExtensions.cs

```diff
@@ -0,0 +1,137 @@
+                    optionsBuilder.UseUpperSnakeCaseNamingConvention(culture);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        public static void NamingConventionsRewriteName(this ModelBuilder modelBuilder,
+            string namingConvention = nameof(DbNamingConvention.Default),
+            CultureInfo culture = null)
+        {
+            if (namingConvention == nameof(DbNamingConvention.Default))
+            {
+                return;
+            }
+
+            if (Enum.TryParse(namingConvention, out DbNamingConvention naming))
+            {
+                NamingConventionsRewriteName(modelBuilder, naming, culture);
+            }
+        }
+
+        public static void NamingConventionsRewriteName(this ModelBuilder modelBuilder,
+            DbNamingConvention namingConvention = DbNamingConvention.Default,
+            CultureInfo culture = null)
+        {
+            if (namingConvention == DbNamingConvention.Default)
+            {
+                return;
+            }
+
+            INameRewriter nameRewriter = null;
+            culture = culture ?? CultureInfo.InvariantCulture;
+            switch (namingConvention)
+            {
+                case DbNamingConvention.Default:
+                    break;
+                case DbNamingConvention.SnakeCase:
+                    nameRewriter = new SnakeCaseNameRewriter(culture);
+                    break;
+                case DbNamingConvention.LowerCase:
+                    nameRewriter = new LowerCaseNameRewriter(culture);
+                    break;
+                case DbNamingConvention.UpperCase:
+                    nameRewriter = new UpperCaseNameRewriter(culture);
+                    break;
+                case DbNamingConvention.UpperSnakeCase:
+                    nameRewriter = new UpperSnakeCaseNameRewriter(culture);
+                    break;
+                default:
+                    break;
+            }
+
+            foreach (var entity in modelBuilder.Model.GetEntityTypes())
+            {
+                entity.SetTableName(nameRewriter.RewriteName(entity.GetTableName()));
+
+                foreach (var property in entity.GetProperties())
+                {
+                    property.SetColumnName(nameRewriter.RewriteName(property.GetColumnName()));
+                }
+
+                foreach (var key in entity.GetKeys())
+                {
+                    key.SetName(nameRewriter.RewriteName(key.GetName()));
+                }
+
+                foreach (var key in entity.GetForeignKeys())
+                {
+                    key.SetConstraintName(nameRewriter.RewriteName(key.GetConstraintName()));
+                }
+
+                foreach (var index in entity.GetIndexes())
+                {
+                    index.SetName(nameRewriter.RewriteName(index.GetName()));
+                }
+            }
+
+        }
```

Comment:
Is it duplicate?
https://github.com/efcore/EFCore.NamingConventions/blob/36a41daa87/EFCore.NamingConventions/Internal/NamingConventionSetPlugin.cs#L15-l40